stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -175,17 +123,29 @@ import SwiftShims
 // ---------------------------------------
 //
 // `Dictionary<K, V>` bridges to `NSDictionary` iff both `K` and `V` are
-// bridged.  Otherwise, a runtime error is raised.
+// bridged.  Otherwise, a runtime error is raised. This bridging is always
+// O(1), but may incur an allocation depending on the following conditions:
 //
-// * if both `K` and `V` are bridged verbatim, then `Dictionary<K, V>` bridges
-//   to `NSDictionary` in `O(1)`, without memory allocation.  Access to
-//   elements does not cause memory allocation.
+// * if both `K` and `V` are bridged verbatim, then `Dictionary<K, V>` is 
+//   toll-free bridged to NSDictionary by returning its 
+//   `_RawNativeDictionaryBuffer`.  However most of the actual implementation 
+//   of `NSDictionary` comes from the `_NativeDictionaryBuffer<K, V>`, which is 
+//   the type most `_RawNativeDictionaryBuffer`s have.
```

Comment:
Yeah I'm maybe trying to stuff too much information in here. I'll clean this up.